compiler/src/lnntoamm/Fn.ts

```diff
@@ -126,76 +120,64 @@ export default class Fn {
     );
   }
 
-  getType(): FunctionType {
-    return this.fnType;
+  acceptsTypes(tys: Type[]): boolean {
+    let params = Object.values(this.params);
+    if (params.length !== tys.length) {
+      return false;
+    }
+    for (let ii = 0; ii < params.length; ii++) {
+      if (!params[ii].ty.compatibleWithConstraint(tys[ii])) {
+        return false;
+      }
+    }
+    return true;
   }
 
   asHandler(amm: Output, event: string) {
-    let handlerArgs = [];
-    for (let arg of Object.keys(this.args)) {
-      handlerArgs.push([arg, this.args[arg].ty.breakdown()]);
+    let handlerParams = [];
+    for (let param of Object.keys(this.params)) {
+      handlerParams.push([param, this.params[param].ty.breakdown()]);
     }
-    amm.addHandler(event, handlerArgs, this.retTy.breakdown());
+    amm.addHandler(event, handlerParams, this.retTy.breakdown());
     let isReturned = false;
     for (let ii = 0; ii < this.body.length; ii++) {
       const stmt = this.body[ii];
-      if (stmt instanceof Dec || stmt instanceof Assign || stmt instanceof Emit) {
-        stmt.inline(amm);
-      } else if (stmt instanceof Exit) {
-        stmt.inline(amm);
+      stmt.inline(amm);
+      if (stmt instanceof Exit) {
         isReturned = true;
         if (ii !== this.body.length - 1) {
           throw new Error(`hmmmm... unreachable statements probably should've been caught earlier?`);
         }
-        break;
-      } else {
-        throw new Error(`did not expect to inline stmt: ${stmt}`);
       }
     }
     if (!isReturned) {
-      if (this.retTy.breakdown() !== opcodes().get('void')) {
+      if (!this.retTy.compatibleWithConstraint(opcodes().get('void'))) {
         throw new Error(`no return value for function`);
```

Comment:
This error message seems misleading? Handlers can't return non-void, and the check appears to be correct, but reading that message I would think I need to return some value for the compiler to be happy.

Reply:
Maybe a message like `Event handlers should not return a value`?